### ISSUE TITLE
"DB::NetException: Connection reset by peer, while writing to socket" on deistributed table queries with LIMIT 

### ISSUE THREAD

**Issue by Brutalis64** [ClickHouse/ClickHouse#10745]
When query distributed table with LIMIT stanza in logs appears error below:
```
2020.05.08 13:36:17.111469 [ 224035 ] {4a69d83e-328e-46f3-a844-07aeb56ab2ef} <Error> executeQuery: Code: 210, e.displayText() = DB::NetException: Connection reset by peer, while writing to socket (10.102.241.49:53684) (version 20.3.8.53 (official build)) (from 10.102.241.49:53684) (in query: SELECT 1 FROM metrics.flink_metrics_local LIMIT 92), Stack trace (when copying this mes
sage, always include the lines below):

0. Poco::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x10542450 in /usr/bin/clickhouse
1. DB::Exception::Exception(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, int) @ 0x8f4272d in /usr/bin/clickhouse
2. ? @ 0x9032396 in /usr/bin/clickhouse
3. DB::WriteBuffer::next() @ 0x8f7c35d in /usr/bin/clickhouse
4. DB::TCPHandler::processOrdinaryQueryWithProcessors() @ 0x9020530 in /usr/bin/clickhouse
5. DB::TCPHandler::runImpl() @ 0x902556b in /usr/bin/clickhouse
6. DB::TCPHandler::run() @ 0x9026470 in /usr/bin/clickhouse
7. Poco::Net::TCPServerConnection::start() @ 0xe3b9a4b in /usr/bin/clickhouse
8. Poco::Net::TCPServerDispatcher::run() @ 0xe3b9ecd in /usr/bin/clickhouse
9. Poco::PooledThread::run() @ 0x105d05b7 in /usr/bin/clickhouse
10. Poco::ThreadImpl::runnableEntry(void*) @ 0x105cc3bc in /usr/bin/clickhouse
11. ? @ 0x105cdd5d in /usr/bin/clickhouse
12. start_thread @ 0x7ea5 in /usr/lib64/libpthread-2.17.so
13. __clone @ 0xfe8cd in /usr/lib64/libc-2.17.so
```
And it seems that client getting correct response data. Error appears on all nodes except that, on which was queried initial request. 
Clickhouse-server version: 20.3.8.53

**Comment by zhbdesign** [comment]
I have the same problem. Have you solved it now?

**Comment by Brutalis64** [comment]
Actually not yet, may be in 20.4, i didn't try it.

**Comment by Ernestyj** [comment]
I also frequently met below error:
<Error> executeQuery: Code: 210, e.displayText() = DB::NetException: I/O error: Broken pipe, while writing to socket

<Error> ServerErrorHandler: Poco::Exception. Code: 1000, e.code() = 104, e.displayText() = Connection reset by peer, Stack trace (when copying this message, always include the lines below):

<Error> InterserverIOHTTPHandler: Poco::Exception. Code: 1000, e.code() = 0, e.displayText() = Not found: endpoint (version 20.4.4.18 (official build))

<Error> executeQuery: Code: 159, e.displayText() = DB::Exception: Timeout exceeded: elapsed 240.045516755 seconds, maximum: 240: While executing MergeTreeThread (version 20.4.4.18 (official build))

Is it just the network issue or something else?

**Comment by John-belt** [comment]
Is this issue resolved ?? i am getting this issue in 20.3 ch version??

**Comment by wangweiA** [comment]
Is this issue resolved ?? i am getting this issue in 20.9.4.76 version??

**Comment by alexey-milovidov** [comment]
This is a diagnostic message about behaviour of the clients and network interaction.
The server correctly informs about connection reset. There are two possibilities:
- if these messages are appeared only occasionally: no action required;
- if these messages appear every time and the system is disrupted: misconfiguration or network issues.
